### PR TITLE
Convert (year specific) Care home into a function for processing

### PR DESCRIPTION
**Pull request by Jennit07**
Addresses #479 and converts two scripts into a function:

` extract_care_homes.R`  and  `extract_care_homes_tests`

I have compared the R version to SPSS and updated the spreadsheet on teams. The output looks good with all variables and the number of rows matching. However there are some differences in beddays and costs compared to the SPSS version. I have added CH to issue #477 for checking other non-matching SPSS vs R outputs.

I have also tidied up some scripts on this branch e.g. HC and AT scripts in the production folder which are not needed as these are already converted into a function.

### CODE REVIEW

File: R/process_extract_care_home.R [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    dplyr::left_join(client_lookup, by = c("sending_location", "social_care_id"))
+
+
+  # Data Cleaning ---------------------------------------
+  source_ch_clean <- ch_data %>%
+    # create variables
+    dplyr::mutate(
+      year = year,
+      recid = "CH",
+      smrtype = "Care-Home"
```
Comment (Moohan) [inline R/process_extract_care_home.R:53]: Implement this in the `smrtype` function, please.

File: R/process_extract_care_home.R [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    ) %>%
+    # compute lca variable from sending_location
+    dplyr::mutate(sc_send_lca = convert_sending_location_to_lca(.data$sending_location)) %>%
+    # bed days
+    create_monthly_beddays(year, .data$record_keydate1, .data$record_keydate2) %>%
+    # year stay
+    dplyr::mutate(
+      yearstay = rowSums(dplyr::across(tidyselect::ends_with("_beddays"))),
+      # total length of stay
+      stay = calculate_stay(year, .data$record_keydate1, .data$record_keydate2, .data$sc_latest_submission)
```
Comment (Moohan) [inline R/process_extract_care_home.R:63]: I think using named parameters in this case (and possibly others) makes it easier to see what's going on.
```suggestion
      stay = calculate_stay(year,
       start_date = .data$record_keydate1,
       end_date =  .data$record_keydate2,
        sc_qtr =  .data$sc_latest_submission)
```

**Review by Moohan — CHANGES_REQUESTED** [review]
I just did a quick first pass. Looking good!

File: R/process_extract_care_home.R (hunk 1 above)
Comment (Jennit07) [reply R/process_extract_care_home.R:53]: Done

File: R/process_extract_care_home.R (hunk 2 above)
Comment (Jennit07) [reply R/process_extract_care_home.R]: Done, also implemented for calculate beddays function

**Comment by Jennit07** [comment]
Changes made, ready for re-review

**Pull request merged**